src/Sylius/Component/Core/Model/Shipment.php

```diff
@@ -44,6 +45,7 @@ public function __construct()
 
     public function getOrder(): ?BaseOrderInterface
     {
+        Assert::nullOrIsInstanceOf($this->order, OrderInterface::class);
```

Comment:
I was wondering for long time, but is it no possible to cover all this cases with extended resource bundle models with phpstan plugin somehow?

Reply:
To clarify the point: There is valid BaseOrderInterface return type hint, but phpstan still forces to check against current OrderInterface.
Might there be an option to make phpstan aware of resource bundle inheritance scheme somehow and skip this assert.

But this is purely off topic.

Reply:
I think this could also be due to some wrong property type. For example, in this class, the order property is annotated as a `BaseOrderInterface|null`, but this is not right, it should be `OrderInterface|null`. Anyway, moving on to PHP 8.0 and 8.1 makes big huge forward in typization these annotations can be changed in a more strict way, but obviously, it is not the purpose of this PR.